### DISCUSSION TITLE
Diagnose urh crass on startup

### DISCUSSION THREAD

**Discussion by jcoles56** [jopohl/urh#878] [Q&A] [answered]
I installed on Linux Mint 20.1 using the "Generic Installation with pip (recommended)" instructions.
When I try to start the app, I get code errors:

$ urh
Traceback (most recent call last):
  File "/usr/local/bin/urh", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/urh/main.py", line 84, in main
    import urh.cythonext.signal_functions
  File "src/urh/cythonext/signal_functions.pyx", line 1, in init urh.cythonext.signal_functions
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject

Does anyone know what the problem might be?

**Comment by jcoles56** [comment, marked as answer]
The fix turned out to be simple.
pip uninstall numpy
pip install numpy

My topic title should say "crash" not "crass", by the way.

**Comment by geo207** [comment]
jcoles56,

When you reinstalled numpy with pip, did it reinstall all of the other programs that were uninstalled when numpy was uninstalled?
When I type pip uninstall numpy, I get the following caution message:

$ pip uninstall numpy
Found existing installation: numpy 1.17.4
Uninstalling numpy-1.17.4:
  Would remove:
    /usr/bin/f2py
    /usr/bin/f2py3
    /usr/bin/f2py3.8
    /usr/lib/python3/dist-packages/numpy
    /usr/lib/python3/dist-packages/numpy-1.17.4.egg-info
Proceed (Y/n)?

I would like to know if your program removed all of these packages and then reinstalled everything back when you reinstalled numpy.  Please also see issue #897:
https://github.com/jopohl/urh/issues/897

**Reply by jcoles56** [reply]
I think it did reinstall those files. I just checked in Synaptic and those files are part of the python3-numpy package, version 1.17.4-5ubuntu3, which is currently installed.

I haven't done much with the URH since installing it. It looks like there's a big learning curve ahead of me.

**Comment by andy778** [comment]
Latest kali linux have similar Expected 96 got 88 from PyObject (numpy)
